src/relations/database_provides.py

```diff
@@ -97,15 +95,34 @@ def create_database_and_user(self, *, router_endpoint: str, shell: mysql_shell.S
         )
         self._set_databag(username=username, password=password, router_endpoint=router_endpoint)
 
+
+class _UserNotCreated(Exception):
+    """Database & user has not been provided to related application charm"""
+
+
+class _CreatedUser(_Relation):
```

Comment:
This feels a little counterintuitive: a `_CreatedUser` is actually a subclass of a `_Relation`

same thing with `_RequestedUser`

Reply:
the name is definitely a big part of it. when i saw it in `reconcile_users`, i was surprised that it was being compared to a `Relation`. i expected it to be a class that encapsulates Users, but it's not quite storing any user info (except a method `delete_user`)